packages/grafana-ui/src/components/TagsInput/TagsInput.tsx

```diff
@@ -1,122 +1,88 @@
-import React, { ChangeEvent, KeyboardEvent, PureComponent } from 'react';
-import { css, cx } from 'emotion';
-import { stylesFactory } from '../../themes/stylesFactory';
+import React, { ChangeEvent, KeyboardEvent, FC, useState } from 'react';
+import { css } from 'emotion';
 import { Button } from '../Button';
-import { Input } from '../Forms/Legacy/Input/Input';
 import { TagItem } from './TagItem';
+import { useStyles } from '../../themes/ThemeContext';
+import { GrafanaTheme } from '@grafana/data';
+import { Input } from '../Input/Input';
 
-interface Props {
+export interface Props {
   placeholder?: string;
   tags?: string[];
-
   onChange: (tags: string[]) => void;
 }
 
-interface State {
-  newTag: string;
-  tags: string[];
-}
-
-export class TagsInput extends PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    this.state = {
-      newTag: '',
-      tags: this.props.tags || [],
-    };
-  }
+export const TagsInput: FC<Props> = ({ placeholder = 'New tag (enter key to add)', tags = [], onChange }) => {
+  const [newTagName, setNewName] = useState<string>('');
+  const styles = useStyles(getStyles);
 
-  onNameChange = (event: ChangeEvent<HTMLInputElement>) => {
-    this.setState({
-      newTag: event.target.value,
-    });
+  const onNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setNewName(event.target.value);
   };
 
-  onRemove = (tagToRemove: string) => {
-    this.setState(
-      (prevState: State) => ({
-        ...prevState,
-        tags: prevState.tags.filter((tag) => tagToRemove !== tag),
-      }),
-      () => this.onChange()
-    );
+  const onRemove = (tagToRemove: string) => {
+    onChange(tags?.filter((x) => x !== tagToRemove));
   };
 
-  // Using React.MouseEvent to avoid tslint error
-  onAdd = (event: React.MouseEvent) => {
+  const onAdd = (event: React.MouseEvent) => {
     event.preventDefault();
-    if (this.state.newTag !== '') {
-      this.setNewTags();
-    }
+    onChange(tags.concat([newTagName]));
```

Comment:
```suggestion
    onChange(tags.concat(newTagName));
```
Don't think this needs to be an array.